### PR TITLE
[Forwardport] 2.3 develop PR port 15942

### PR DESCRIPTION
**Pull request by eduard13**
Original Pull Request: https://github.com/magento/magento2/pull/15942

### Fixed Issues (if relevant)
1. magento/magento2#13747: Wysiwyg > Image Uploader >Max width/height

### Manual testing scenarios
Check the task's scenarios

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @eduard13. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on Pull Request changes
- `@magento-engcom-team give me new test instance` - deploy **NEW** test instance based on Pull Request changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy Vanilla Magento instance for Issue or Pull Request

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

File: app/code/Magento/Backend/Block/Media/Uploader.php [hunk 1]
```diff
@@ -44,11 +51,13 @@ class Uploader extends \Magento\Backend\Block\Widget
     public function __construct(
         \Magento\Backend\Block\Template\Context $context,
         \Magento\Framework\File\Size $fileSize,
+        \Magento\Framework\Image\Adapter\ConfigInterface $imageConfig,
```
Comment (VladimirZaets) [inline app/code/Magento/Backend/Block/Media/Uploader.php]: Hi, @eduard13, due to [Magento backward-compatible guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) we can't add new required parameters to the constructor. 

Please make this param optional, the example you can look [here](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/)

File: app/code/Magento/Backend/Block/Media/Uploader.php [hunk 2]
```diff
@@ -30,6 +32,11 @@ class Uploader extends \Magento\Backend\Block\Widget
      */
     protected $_fileSizeService;
 
+    /**
+    * @var \Magento\Framework\Image\Adapter\ConfigInterface
+    */
+    protected $imageConfig;
```
Comment (VladimirZaets) [inline app/code/Magento/Backend/Block/Media/Uploader.php]: Due to [Magento backward-compatible guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) we can't add protected properties. All new properties should be private.

**Comment by eduard13** [comment]
@VladimirZaets thank you for the reviews. I've fixed them, moreover I've added also some other changes, which are fixing some modularity perspectives.
Also I'm not sure what should I do within the failing tests, can you please have a look and advice me something here?

**Pull request closed without merging**